notes-0225-universes.tex

```diff
@@ -0,0 +1,272 @@
+\newcommand{\isprop}[1]{\ensuremath{\textnormal{is-prop}(#1)}}
+\newcommand{\iscontr}[1]{\ensuremath{\textnormal{is-contr}(#1)}}
+\newcommand{\ishae}[1]{\ensuremath{\textnormal{is-hae}(#1)}}
+\newcommand{\ap}{\textnormal{ap}}
+\newcommand{\idtoequiv}{\textnormal{idtoequiv}}
+\newcommand{\funapp}{\textnormal{funapp}}
+
+\title{CSCI 8980 Higher-Dimensional Type Theory\\ Lecture Notes}
+\author{Dawn Michaelson, Jack Perisich}
+\date{February 25, 2020}
+
+\begin{document}
+
+\maketitle
+
+
+
+
+
+\section{Universe}
+To internalize the $\istype{\Gamma}{A}$ judgment, we introduce the
+universe type \U.  This can be viewed as the ``type of types''.  We
+replace the original judgment with $\oftype{\Gamma}{A}{\U}$.
+%
+We can make this replacement in all our existing rules.  For example,
+we originally had the following rule for dependent function types:
+\begin{prooftree*}
+  \hypo{\istype{\Gamma}{A}}
+  \hypo{\istype{\Gamma, x:A}{B}}
+  \infer2{\istype{\Gamma}{\dfuntype{x}{A}{B}}}
+\end{prooftree*}
+We can replace each judgment with a judgment that the given type has
+the type of the universe, giving us a new rule:
+\begin{prooftree*}
+  \hypo{\oftype{\Gamma}{A}{\U}}
+  \hypo{\oftype{\Gamma, x:A}{B}{\U}}
+  \infer2{\oftype{\Gamma}{\dfuntype{x}{A}{B}}{\U}}
+\end{prooftree*}
+Doing this in all rules gives a system where we do not need the
+original is-type judgment.
+
+
+
+
+
+\section{Inconsistency}
+Because \U{} is a type, we should have a rule:
+\begin{prooftree*}
+  \infer0{\oftype{\Gamma}{\U}{\U}}
+\end{prooftree*}
+With this rule, we can prove inconsistency, meaning a closed term of
+the empty type.  This is the \textbf{Burali-Forti Paradox}.
+\begin{enumerate}
+\item Consider types with a transitive, well-founded order.  We will
+  write this as \wf{A}.  This means that the type has an induction
+  principle and a minimal element.
+\item We define a type $W := \dpairtype{A}{\U}{\wf{A}}$.  This is a
+  type which contains all well-founded types.
+\item This type $W$ is well-founded with embedding as the relation.
+  \begin{itemize}
+  \item An embedding is an order-preserving function with a strict
+    upper bound.  If we have types with ordering relations $(A, <_A)$
+    and $(B, <_B)$ with an embedding $\oftype{f}{A \rightarrow B}$,
+    $f$ will map ordered elements from $A$ to the same order in $B$,
+    and there is some $b$ such that $f(a) <_B b$ for any $a$.
+  \end{itemize}
+  We can prove an embedding is transitive and well-founded.  Then we
+  have $\oftype{(W, \textnormal{embedding})}{W}$.
+\item Every well-founded type $(A, <_A)$ can be embedded into $W$.  We
+  can define a map $a \mapsto \dpairtype{a'}{A}{a' <_A a}$.
+\item Because of the previous point, $W$ can then be embedded into
+  itself, yielding $W <_W W$.  This contradicts that embedding is
+  well-founded.
+\end{enumerate}
+This was originally done in type theory by Girard.  It was later
+improved by Coquand and Hurken.
```

Comment:
Please cite the papers:
- Interprétation fonctionnelle et Élimination des coupure de l'arithmétique d'ordre supérieur by Jean-Yves Girard
- An Analysis of Girard's Paradox by Thierry Coquand
- A Simpliﬁcation of Girard's Paradox by Antonius J.C. Hurken

Reply:
I couldn't find any more about the paper by Girard than its publication year.  Do you have more information about it?

Reply:
It's his Ph.D. thesis (or equivalent) from Université Paris VII. I don't have a copy. :-/